### DISCUSSION TITLE
Support for fetching (non-nested) Links at an arbitrary document layer beyond the 2nd

### DISCUSSION THREAD

**Discussion by TLeitzbach** [BeanieODM/beanie#457] [Feature Request]
Hi @roman-right,

I think it would be great if beanie could support the fetching of linked documents at an arbitrary depth even if they are not nested. 
The latter is available since 1.16.0 and I think it would be a relatively quick gain with great usability.

E.g. with beanie == 1.16.4

```python
from beanie import Document, Link


class LinkedDocument(Document):
    some_prop: int   

class DocumentWithLink(Document):
    my_id: int
    linked_doc: Link[LinkedDocument]

class SomeDocument(Document):
    prop_1: str
    nice_doc_prop: DocumentWithLink

...

await SomeDocument.find_one(
   SomeDocument.prop_1 == 'my_prop', fetch_links=True
)
```

Let me know what you think and many thanks!

**Comment by roman-right** [comment]
Hey!
Sorry for not responding. Yes, this would be great. Handling links inside not linked documents or even BaseModels would be a great feature. I'm thinking about how to make it efficient now.
@noaHson86 works on this also, as I know. Right?

**Reply by noaHson86** [reply]
i do and i might give you an update on that on sat/sun.

**Reply by TLeitzbach** [reply]
Very cool, thank you very much! Let me know if I can helpt. Best:)

**Reply by vvuk** [reply]
Old discussion! Is there any update on this? Specifically, supporting links inside nested BaseModels (not Documents) inside a document? e.g.:

```
class Foo(Document):
  value: int

class Bar(BaseModel):
  foo: Link[Foo]

class TopLevel(Document):
  b0: Bar
  b1: Bar
```

the `foo` links don't seem to be properly handled. (At least, `fetch_links` doesn't seem to have any impact either positive or negative on them)

**Comment by staticxterm** [comment]
Hi @vvuk , this feature was not really fully supported. PRs are welcome.
